blobs/howto/python/blob-devguide-py/blob-devguide-upload-async.py

```diff
@@ -0,0 +1,119 @@
+
+    # <Snippet_upload_blob_data>
+    async def upload_blob_data(self, blob_service_client: BlobServiceClient, container_name: str):
+        blob_client = blob_service_client.get_blob_client(container=container_name, blob="sample-blob.txt")
+        data = b"Sample data for blob"
+
+        # Upload the blob data - default blob type is BlockBlob
+        await blob_client.upload_blob(data, blob_type="BlockBlob")
+    # </Snippet_upload_blob_data>
+
+    # <Snippet_upload_blob_stream>        
+    async def upload_blob_stream(self, blob_service_client: BlobServiceClient, container_name: str):
+        blob_client = blob_service_client.get_blob_client(container=container_name, blob="sample-blob.txt")
+        input_stream = io.BytesIO(os.urandom(15))
+        await blob_client.upload_blob(input_stream, blob_type="BlockBlob")
+    # </Snippet_upload_blob_stream>
+
+    # <Snippet_upload_blob_file>
+    async def upload_blob_file(self, blob_service_client: BlobServiceClient, container_name: str):
+        container_client = blob_service_client.get_container_client(container=container_name)
+        with open(file=os.path.join('filepath', 'filename'), mode="rb") as data:
+            blob_client = await container_client.upload_blob(name="sample-blob.txt", data=data, overwrite=True)
+    # </Snippet_upload_blob_file>
+
+    # <Snippet_upload_blob_tags>
+    async def upload_blob_tags(self, blob_service_client: BlobServiceClient, container_name: str):
+        container_client = blob_service_client.get_container_client(container=container_name)
+        sample_tags = {"Content": "image", "Date": "2022-01-01"}
+        with open(file=os.path.join('filepath', 'filename'), mode="rb") as data:
+            blob_client = await container_client.upload_blob(name="sample-blob.txt", data=data, tags=sample_tags)
+    # </Snippet_upload_blob_tags>
+
+    # <Snippet_upload_blob_blocks>
+    async def upload_blocks(self, blob_container_client: ContainerClient, local_file_path: str, block_size: int):
+        file_name = os.path.basename(local_file_path)
+        blob_client = blob_container_client.get_blob_client(file_name)
+
+        with open(file=local_file_path, mode="rb") as file_stream:
+            block_id_list = []
+
+            while True:
+                buffer = await file_stream.read(block_size)
+                if not buffer:
+                    break
+
+                block_id = uuid.uuid4().hex
+                block_id_list.append(BlobBlock(block_id=block_id))
+
+                await blob_client.stage_block(block_id=block_id, data=buffer, length=len(buffer))
+
+            await blob_client.commit_block_list(block_id_list)
+    # </Snippet_upload_blob_blocks>
+
+    # <Snippet_upload_blob_transfer_options>
+    async def upload_blob_transfer_options(self, account_url: str, container_name: str, blob_name: str):
+        # Create a BlobClient object with data transfer options for upload
+        blob_client = BlobClient(
```

Comment:
Since you are not creating this client from the original service client, you should probably use an `async with` here.